zk/l1infotree/updater_test.go

```diff
@@ -0,0 +1,206 @@
+	"github.com/ledgerwatch/erigon/zk/hermez_db"
+	"github.com/ledgerwatch/erigon/zk/l1infotree"
+	"github.com/ledgerwatch/erigon/zk/syncer"
+	"github.com/stretchr/testify/assert"
+	"go.uber.org/mock/gomock"
+)
+
+// to be removed
+func GetDbTx() (tx kv.RwTx, cleanup func()) {
+	dbi, err := mdbx.NewTemporaryMdbx(context.Background(), "")
+	if err != nil {
+		panic(err)
+	}
+	tx, err = dbi.BeginRw(context.Background())
+	if err != nil {
+		panic(err)
+	}
+
+	err = hermez_db.CreateHermezBuckets(tx)
+	if err != nil {
+		panic(err)
+	}
+
+	return tx, func() {
+		tx.Rollback()
+		dbi.Close()
+	}
+}
+
+func TestNewHermezDb(t *testing.T) {
+	tx, cleanup := GetDbTx()
+	defer cleanup()
+	db := hermez_db.NewHermezDb(tx)
+	assert.NotNil(t, db)
+}
+
+func TestNewUpdater(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	cfg := &ethconfig.Zk{}
+
+	l1InfoTreeSyncer := syncer.NewL1Syncer(
+		ctx,
+		nil, nil, nil,
+		cfg.L1BlockRange,
+		cfg.L1QueryDelay,
+		cfg.L1HighestBlockType,
+	)
+
+	updater := l1infotree.NewUpdater(cfg, l1InfoTreeSyncer)
+	assert.NotNil(t, updater)
+}
+
+func TestUpdater_WarmUp(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+	EthermanMock := mocks.NewMockIEtherman(mockCtrl)
+
+	header := &types.Header{Number: big.NewInt(1), Difficulty: big.NewInt(100)}
+	expectedBlock := types.NewBlockWithHeader(header)
+
+	EthermanMock.EXPECT().
+		BlockByNumber(gomock.Any(), (*big.Int)(nil)).
+		Return(expectedBlock, nil).
+		AnyTimes()
+
+	cfg := &ethconfig.Zk{}
+
+	l1InfoTreeSyncer := syncer.NewL1Syncer(
+		ctx,
+		[]syncer.IEtherman{EthermanMock},
+		[]common.Address{},
+		[][]common.Hash{},
+		cfg.L1BlockRange,
+		cfg.L1QueryDelay,
+		cfg.L1HighestBlockType,
+	)
+
+	updater := l1infotree.NewUpdater(cfg, l1InfoTreeSyncer)
+
+	tx, cleanup := GetDbTx()
+	defer cleanup()
+
+	err := updater.WarmUp(tx)
+	// log error
+	log.Println(err)
+	assert.NoError(t, err)
+}
+
+func TestUpdater_GetProgress(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	cfg := &ethconfig.Zk{}
+
+	l1InfoTreeSyncer := syncer.NewL1Syncer(
+		ctx,
+		nil, nil, nil,
+		cfg.L1BlockRange,
+		cfg.L1QueryDelay,
+		cfg.L1HighestBlockType,
+	)
+
+	updater := l1infotree.NewUpdater(cfg, l1InfoTreeSyncer)
+	assert.NotNil(t, updater)
+
+	progress := updater.GetProgress()
+	assert.Equal(t, uint64(0), progress)
+}
+
+func TestUpdater_GetLatestUpdate(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	cfg := &ethconfig.Zk{}
+
+	l1InfoTreeSyncer := syncer.NewL1Syncer(
+		ctx,
+		nil, nil, nil,
+		cfg.L1BlockRange,
+		cfg.L1QueryDelay,
+		cfg.L1HighestBlockType,
+	)
+
+	updater := l1infotree.NewUpdater(cfg, l1InfoTreeSyncer)
+	assert.NotNil(t, updater)
+
+	latestUpdate := updater.GetLatestUpdate()
+	assert.Nil(t, latestUpdate)
```

Comment:
Disagree, because we are only testing the warmup.

what we could do is add a warm up to the other tests ! ... I will lot into it. not sure, I'm not 100% up to date in what we wanted to do with all this code haha